File: Phase1/test/TokenizerTest.java
package test;

import src.*;
import java.util.*;
//import static org.junit.jupiter.api.Assertions.assertEquals;
//import org.junit.jupiter.api.Test;
import static org.junit.Assert.*;
import org.junit.Test;
import org.junit.Before;

public class TokenizerTest {
    @Test
    public void testTokenizer() {
        HashSet<String> expected = new HashSet<>(Arrays.asList("hello","hi", "java", "python", "how", "test"));
        String testString = "Hello-hi, how? java*python! test.";
        Tokenizer tokenizer = new Tokenizer();
        assertEquals(expected, tokenizer.tokenize(testString));
    }
}

File: Phase08/SearchTest/TokenizerTest.cs
﻿using System.Collections.Generic;
using Search;
using Search.Interface;
using Xunit;

namespace SearchTest
{
    public class TokenizerTest
    {
        private readonly ITokenizer _tokenizer;
        public TokenizerTest()
        {
            _tokenizer = new Tokenizer();
        }

        [Fact]
        public void TestTokenizeMethod_ShouldExtractWords()
        {
            const string testString = "Hello-hi, how? java*python! test.";
            var expected = new HashSet<string>(){"hello", "hi", "how","java","python","test"};
            Assert.Equal(expected, _tokenizer.Tokenize(testString));
        }
    }
}